### PR TITLE
I add the code which is support Micro$oft Visual Studio 2010

### PR DESCRIPTION
**Pull request by holsetychen**

### CODE REVIEW

**Comment by imZack** [comment]
wow, looks like we are M$ compatible now :+1:

**Comment by ckgczarzar** [comment]
Good job.
You are the real Sanji user now.

File: src/unit_test/resource_unit_test.c [hunk 1]
```diff
@@ -4,6 +4,23 @@ int main()
...
+	char m2[RESOURCE_METHOD_LEN] = "GET";
+	char m3[RESOURCE_METHOD_LEN] = "PUT";
+	char m4[RESOURCE_METHOD_LEN] = "DELETE";
+	char m5[RESOURCE_METHOD_LEN] = "POS";
+	char m6[RESOURCE_METHOD_LEN] = "PO";
+	char m7[RESOURCE_METHOD_LEN] = "PPOST";
+	char m8[RESOURCE_METHOD_LEN] = "PPPOST";
+	char m9[RESOURCE_METHOD_LEN] = "POSTT";
+	char m10[RESOURCE_METHOD_LEN] = "POSTTT";
+    int ret;
```
Comment (lwindg) [inline src/unit_test/resource_unit_test.c:18]: tab should be used instead

File: src/unit_test/component_unit_test.c [hunk 2]
```diff
@@ -5,6 +5,10 @@ int main()
 	struct component *component = NULL;
 	struct component *node = NULL;
 	int i;
+    int ret;
+    char hook[4][COMPONENT_NAME_LEN];
```
Comment (lwindg) [inline src/unit_test/component_unit_test.c:9]: tab should be used instead

File: src/unit_test/component_unit_test.c [hunk 3]
```diff
@@ -150,9 +152,7 @@ int main()
 	component_add_node(component, "n_ccc", NULL, "t_ccc", "r_ccc", "n_aaa", 1, 333, 0);
 	component_display(component);
 
-	char *names = NULL;
-	unsigned int names_count;
-	names = component_get_names_by_hook(component, "n_aaa", &names_count);
+    names = component_get_names_by_hook(component, "n_aaa", &names_count);
```
Comment (lwindg) [inline src/unit_test/component_unit_test.c:155]: tab should be used instead

**Pull request closed without merging**